subsys/net/lib/aws_iot/Kconfig

```diff
@@ -0,0 +1,75 @@
+#
+# Copyright (c) 2019 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+menuconfig AWS_IOT
+	bool "AWS IoT library"
+	select CLOUD_API
+	select MQTT_LIB
+	select MQTT_LIB_TLS
+	select AT_CMD
+
+if AWS_IOT
+
+config AWS_IOT_STATIC_IPV4
+	bool "Enable use of static IPv4"
+
+config AWS_IOT_STATIC_IPV4_ADDR
+	string "Static IPv4 address"
+	depends on AWS_IOT_STATIC_IPV4
+
+config AWS_IOT_SEC_TAG
+	int "Security tag to use for AWS IoT connection"
+	default 42
+
+config AWS_IOT_HOST_NAME
+	string "AWS IoT server hostname"
+	default "a34x44yyrk96tg-ats.iot.eu-central-1.amazonaws.com"
+
+config AWS_IOT_PORT
+	int "AWS server port"
+	default 8883
+
+config AWS_IOT_MQTT_RX_TX_BUFFER_LEN
+	int "Buffer sizes for the MQTT library."
+	default 2048
+	help
+	  Specifies maximum message size can be transmitted/received through
+	  MQTT (exluding MQTT PUBLISH payload).
```

Comment:
Is the required buffer size that big if the PUBLISH payload is excluded? By the way, does PUBLISH in this context mean the payload received in an `MQTT_PUBLISH` event, or what's sent in a publish message?

Reply:
The payload buffer is the buffer populated during a ```PUBLISH``` event yes. So receiving.

Reply:
Im not sure I understand the first question there.

Reply:
I was referring to the help text: `... (exluding MQTT PUBLISH payload)`. If the publish payload is not supposed to go into this buffer, then I assume it's intended for storing the MQTT header and for example payload in a `SUBSCRIBE` message. In that case, default of 2048 seems excessive.